src/main/java/com/google/starfish/servlets/UserRegistrationServlet.java

```diff
@@ -0,0 +1,170 @@
+      // Set an error code of 500 if the server can't connect to the database
+      res.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      res.getWriter().println("INTERNAL SERVER ERROR");
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance())
+      .setAudience(Collections.singletonList(CLIENT_ID))
+      .build();
+    try {
+      GoogleIdToken idToken = verifier.verify(req.getParameter("idToken"));
+      if (idToken == null) {
+        res.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        res.getWriter().println("INVALID TOKEN ID");
+        return;
+      }
+
+      Payload payload = idToken.getPayload();
+
+      String userId = payload.getSubject();
+
+      // Get profile information from payload
+      String email = payload.getEmail();
+      String name = (String) payload.get("name");
+      String pictureUrl = (String) payload.get("picture");
+      String givenName = (String) payload.get("given_name");
+
+      DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");
+
+      try (Connection conn = pool.getConnection()) {
+        boolean userExists = checkIfUserExists(conn, userId);
+        // Do nothing if the user already exists
+        if (!userExists) {
+          String stmt =
+              "INSERT INTO users ( "
+                  + "id,"
+                  + "display_picture,"
+                  + "display_name,"
+                  + "date_joined,"
+                  + "email,"
+                  + "points,"
+                  + "school ) "
+            + "VALUES ( "
+                  + "?,"
+                  + "?,"
+                  + "?,"
+                  + "?,"
+                  + "?,"
+                  + "?,"
+                  + "? ); ";
+          try (PreparedStatement userStmt = conn.prepareStatement(stmt)) {
+            userStmt.setString(1, userId);
+            userStmt.setString(2, pictureUrl);
+            userStmt.setString(3, givenName);
+            userStmt.setDate(4, new Date(Calendar.getInstance().getTimeInMillis()));
+            userStmt.setString(5, email);
+            // User starts with 0 points
+            userStmt.setInt(6, 0);
+            userStmt.setNull(7, Types.VARCHAR);
+
+            // Finally, execute the statement. If it fails, an error will be thrown
+            userStmt.execute();
+
+          }
+        }
+
+        // Create new user session and save as cookie on client side
+        HttpSession newSession = req.getSession(true);
+        newSession.setAttribute("user_id", userId);
+        Cookie activeSession = new Cookie("SFCookie", newSession.getId());
+        res.addCookie(activeSession);
+
+        // TO-DO: Once we have a UserService, we can create a User object and send that back to client
```

Comment:
Does this mean that after making a POST request to user-signin servlet, it won't be necessary to make a separate GET request to retrieve the User object?

Reply:
Oh good point, I must have forgotten that that's what we'd talked about when I wrote this comment. I think you're still right in that the POST and then GET flow is what makes most sense here, I'll update this comment to reflect that.